src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2738,6 +2738,22 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 }
             }
 
+            if (operand.IsLiteralDefault())
+            {
+                // PROTOTYPE(default) Need to handle type parameters here (`where T : struct` for instance)
+                //if (targetType.TypeKind == TypeKind.TypeParameter)
+                //{
+                //    Error(diagnostics, ErrorCode.ERR_TypeVarCantBeNull, syntax, targetType);
+                //    return;
+                //}
+
+                if (!targetType.IsStructType() || targetType.IsNullableType())
```

Comment:
Whether `default` should be allowed on reference types is listed as an open LDM question. But the more I discuss with folks and the clearer it is that folks want to allow it. I'll remove this check.